internal/pkg/gateway/commit/blockparser.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2021 IBM All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package commit
+
+import (
+	"github.com/golang/protobuf/proto"
+	commonProto "github.com/hyperledger/fabric-protos-go/common"
+	peerProto "github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/pkg/errors"
+)
+
+type blockParser struct {
```

Comment:
You're right, there is more than one way this could be implemented, and plain functions is one. That would require a couple of functions and their calls to have an additional `Block` parameter. I find it nice to avoid that for this self-contained set of inter-related functions. Functions with fewer parameters are generally considered more readable and maintainable

Reply:
+1, I won't argue that this be changed, but it's odd stylistically to me to create a structure with no state, just to implicitly pass a parameter around.

Reply:
It's upsetting enough people, I'll change it